internal/services/dynatrace/dynatrace_monitors_resource.go

```diff
@@ -270,33 +272,17 @@ func expandDynatraceIdentity(input []identity.ModelSystemAssignedUserAssigned) (
 		Type: monitors.ManagedIdentityType(config.Type),
 	}
 
-	if string(config.Type) == string(monitors.ManagedIdentityTypeUserAssigned) || string(config.Type) == string(monitors.ManagedIdentityTypeSystemAndUserAssigned) {
-		userAssignedIdentities := make(map[string]monitors.UserAssignedIdentity)
-		for k := range config.IdentityIds {
-			userAssignedIdentities[k] = monitors.UserAssignedIdentity{
-				ClientId:    pointer.From(config.IdentityIds[k].ClientId),
-				PrincipalId: pointer.From(config.IdentityIds[k].PrincipalId),
-			}
-		}
-		dynatraceIdentity.UserAssignedIdentities = pointer.To(userAssignedIdentities)
-	}
 	return &dynatraceIdentity, nil
 }
 
-func flattenDynatraceIdentity(input *monitors.IdentityProperties) ([]identity.ModelSystemAssignedUserAssigned, error) {
+func flattenDynatraceIdentity(input *monitors.IdentityProperties) ([]identity.ModelSystemAssigned, error) {
 	if input == nil {
 		return nil, fmt.Errorf("flattening Dynatrace identity: input is nil")
 	}
 
-	var ids []string
-	for v := range pointer.From(input.UserAssignedIdentities) {
-		ids = append(ids, v)
-	}
-
-	return []identity.ModelSystemAssignedUserAssigned{
+	return []identity.ModelSystemAssigned{
 		{
-			Type:        identity.Type(input.Type),
-			IdentityIds: ids,
+			Type: identity.Type(input.Type),
```

Comment:
Isn't the `PrincipalId` and `TenantId` are also needed to be flattend here (also need be defined in the schema)?